pkg/nad/masterplugin.go

```diff
@@ -393,3 +393,161 @@ func (plugin *MasterIPVlanPlugin) GetMasterPluginConfig() (*MasterPlugin, error)
 
 	return plugin.masterPlugin, nil
 }
+
+// MasterBondPlugin provides struct for MasterPlugin to create a NAD for bonded interfaces.
+type MasterBondPlugin struct {
+	masterPlugin *MasterPlugin
+	errorMsg     string
+}
+
+// NewMasterBondPlugin define network-attachment-interface for bond interface.
+func NewMasterBondPlugin(name string) *MasterBondPlugin {
+	glog.V(100).Infof(
+		"Initializing new MasterBondPlugin structure %s", name)
+
+	builder := MasterBondPlugin{
+		masterPlugin: &MasterPlugin{
+			CniVersion: "0.3.1",
+			Name:       name,
+			Type:       "bond",
+		},
+	}
+
+	if builder.masterPlugin.Name == "" {
+		glog.V(100).Infof("error NewMasterBondPlugin name can not be empty")
+
+		builder.errorMsg = "NewMasterBondPlugin name is empty"
+	}
+
+	return &builder
+}
+
+// WithMode defines and sets the bond mode type.
+func (plugin *MasterBondPlugin) WithMode(mode string) *MasterBondPlugin {
+	validModes := map[string]bool{
+		"balance-rr":    true,
+		"active-backup": true,
+		"balance-xor":   true,
+	}
+
+	if mode == "" {
+		glog.V(100).Infof("error adding empty mode to MasterBondPlugin")
+
+		return nil
+	}
+
+	// Check if the provided mode is valid
+	if !validModes[mode] {
+		glog.V(100).Infof("error: invalid mode type used for MasterBondPlugin bond interface")
+
+		return nil
+	}
+
+	plugin.masterPlugin.Mode = mode
+
+	return plugin
+}
+
+// WithLinksInContainer defines nad with linksInContainer as true.
+func (plugin *MasterBondPlugin) WithLinksInContainer(linksInContainer bool) *MasterBondPlugin {
+	if !linksInContainer {
+		glog.V(100).Infof("error adding empty linksInContainer to MasterBondPlugin")
+
+		plugin.errorMsg = "error adding empty linksInContainer to MasterBondPlugin"
+	}
+
+	plugin.masterPlugin.LinksInContainer = linksInContainer
+
+	return plugin
+}
+
+// WithFailOverMac defines nad with WithFailOverMac amount.
+func (plugin *MasterBondPlugin) WithFailOverMac(failOverMac int) *MasterBondPlugin {
+	if failOverMac == 0 {
+		glog.V(100).Infof("error adding empty FailOverMac to MasterBondPlugin")
+
+		plugin.errorMsg = "error adding empty FailOverMac to MasterBondPlugin"
+	}
+
+	plugin.masterPlugin.FailOverMac = failOverMac
+
+	return plugin
+}
+
+// WithMiimon defines nad with WithMiimon.
+func (plugin *MasterBondPlugin) WithMiimon(miimon string) *MasterBondPlugin {
+	if miimon == "" {
```

Comment:
maybe we need add validation that miimon is integer (not random srtring) and is higher than 0